File: patient/patient.py
from flask import Blueprint,render_template,request
import json
import patient.patient_model as patient_model
patient=Blueprint("patients",__name__,url_prefix="/patients")

#page Loaders
@patient.route("/")
@patient.route("/allPatients")
def allpatients():
    return render_template("allpatients.html")

@patient.route("/appconfig")
def appconfig():
    return render_template("appconfig.html")

@patient.route("/privilegegroups")
def privilegegroups():
    return render_template("privilegegroups.html")


#operations

@patient.route("/addpatient",methods=["POST"])
def addPatient():
    userData=request.json
    response={"data":patient_model.add(userData["name"],userData["email"],userData["phone"],userData["gender"],userData["bloodGroup"],userData["medicalHistory"],userData["emergencyNumber"],userData["age"]),"message":"patient Added successfully !"}
    return json.dumps(response)





@patient.route("/updatePatient",methods=["POST"])
def updatePatient():
    userData=request.json
    #id,name,phone,email,age,gender,blood_group,medical_history,emergency_number
    response={"data":patient_model.update(userData["id"],userData["name"],userData["phone"],userData["email"],userData["gender"],userData["age"],userData["bloodGroup"],userData["medicalHistory"],userData["emergencyNumber"]),"message":"patient Updated successfully !"}
    print(userData)
    return json.dumps(response)

@patient.route("/getAllpatients",methods=['GET'])
def getAllpatients():
    try:
        response=patient_model.getAllPatients()
        return json.dumps(response)
    except Exception as error:
        response = {"message":error}
        return json.dumps(response) 
        

@patient.route("/getpatientDetails",methods=['POST'])
def getpatientDetails():
    try:
        data=request.json
        response={"data":patient_model.get(data["id"]),"message":"patient Data Fetched Successfully !"}
        return json.dumps(response)
    except Exception as e:
        print(str(e))
        return False

@patient.route("/disablepatient",methods=['POST'])
def disablepatient():
    return "Disable patient"

@patient.route("/enablepatient",methods=['POST'])
def enablepatient():
    return "Enable patient"

@patient.route("/permanentlyDeletepatient",methods=['POST'])
def permanentlyDeletepatient():
    return "Delete  patient"

File: templates/includes/assets/js/app/patient.js
var app = angular.module('myApp', []).config(['$interpolateProvider', function ($interpolateProvider) {
    $interpolateProvider.startSymbol('{a');
    $interpolateProvider.endSymbol('a}');
}]);

app.controller('PatientController', ['$scope', '$http', function ($scope, $http) {
    $scope.action = 0; // 0 = ADD , 1 = UPDATE , 2 = DELETE 
    $scope.selectedpatientiD = 0;
    $scope.patients = "";
    $scope.patientName = "";
    $scope.patientEmail = "";
    $scope.patientPhone = "";
    $scope.patientGender = "";
    $scope.patientBloodGroup = 1;
    $scope.patientAge = 0;
    $scope.patientMedicalHistory = "";
    $scope.patientEmergencyNumber="";
    $scope.bloodGroups = [{"id":1,"name":"A"},{"id":2,"name":"B"},{"id":3,"name":"AB"},{"id":4,"name":"O"}];
    postdata = "";
    $http.get("patients/getAllpatients").then(function (response) {
        $scope.patients = response.data;
    });
    $scope.loadModal = function (action) {
        $scope.action = 1;
        $("#entity_modal").modal('show');



    }
    $scope.getDetails = function () {
        $scope.action = 2;
        var newObj = new Object();
        newObj.id = $scope.selectedpatientiD;
        postdata = JSON.stringify(newObj);
        console.log(postdata);
        var post = $http({
            method: "POST",
            url: "patients/getpatientDetails",
            dataType: 'json',
            data: postdata,
            headers: { 'Content-Type': 'application/json' }
        }).then(function successCallback(response) {
            var patient = response.data.data;
            console.log(patient);

            $scope.patientName = patient[1];
            $scope.patientEmail = patient[2];
            $scope.patientPhone = patient[3];
            $scope.patientAge = patient[4];
            $scope.patientGender = patient[5];
            $scope.patientBloodGroup = patient[6];
            $scope.patientMedicalHistory = patient[7];
            $scope.patientEmergencyNumber = patient[8];

            $("#entity_modal").modal('show');


        }, function errorCallback(response) {


        });




    }

    $scope.selectPatient = function (id) {
        $scope.selectedpatientiD = id;
    }
    $scope.getAllPatients = function(){
        $http.get("patients/getAllpatients").then(function (response) {
            $scope.patients = [];
            $scope.patients = response.data;
        });

    };

    $scope.submitForm = function () {
        var newObj = new Object();
        newObj.id = $scope.selectedpatientiD;
        newObj.name = $scope.patientName;
        newObj.email = $scope.patientEmail;
        newObj.phone = $scope.patientPhone;
        newObj.gender = $scope.patientGender;
        newObj.bloodGroup = $scope.patientBloodGroup;
        newObj.medicalHistory = $scope.patientMedicalHistory;
        newObj.emergencyNumber = $scope.patientEmergencyNumber
        newObj.age=$scope.patientAge;
        postdata = JSON.stringify(newObj);
        switch ($scope.action) {
            case 1:
                // Add Code Here
                var post = $http({
                    method: "POST",
                    url: "patients/addpatient",
                    dataType: 'json',
                    data: postdata,
                    headers: { 'Content-Type': 'application/json' }
                }).then(function successCallback(response) {
                    showSuccessMessage("patient " + $scope.patientName + " Details Added Sucessfully ! ");
                    $("#entity_modal").modal('hide');


                }, function errorCallback(response) {


                });
                break;
            case 2:
                //Update Code Here 
                var post = $http({
                    method: "POST",
                    url: "patients/updatePatient",
                    dataType: 'json',
                    data: postdata,
                    headers: { 'Content-Type': 'application/json' }
                }).then(function successCallback(response) {
                    showSuccessMessage("patient " + $scope.patientName + " Details Updated Sucessfully ! ");
                    $("#entity_modal").modal('hide');
                    console.log(newObj.id);


                }, function errorCallback(response) {


                });
                break;
        }
        $scope.getAllPatients();
    }

}]);